### DISCUSSION TITLE
How to install and use Whisper offline (no internet required)

### DISCUSSION THREAD

**Discussion by nicholasgcotton** [openai/whisper#1463] [Show and tell]
This guide can also be found at [Whisper Full (& Offline) Install Process for Windows 10/11](https://github.com/nicholasgcotton/WhisperDO/blob/main/Whisper%20Install%20(Offline).md)

# Whisper Full (& Offline) Install Process for Windows 10/11 
Purpose: These instructions cover the steps not explicitly set out on the main [Whisper](https://github.com/openai/whisper) page, e.g. for those who have never used python code/apps before and do not have the prerequisite software already installed. 

## Requirements: 
- Full admin rights on your computer.
- A PC with a CUDA-capable dedicated GPU with at least 4GB of VRAM (but more VRAM is better).  See: [Available models and languages](https://github.com/openai/whisper#available-models-and-languages)
- For online installation: An Internet connection for the initial download and setup.
- For offline installation: Download on another computer and then install manually using the "OPTIONAL/OFFLINE" instructions below.

### Step 1: Unlisted Pre-Requisites
- Before you can run whisper you must download and install the follopwing items. (For offline installation just download the files on another machine and move them to your offline machine to install them.) 
    - NVIDIA CUDA drivers: https://developer.nvidia.com/cuda-downloads 
    -  Python 3.9 or 3.10 (x64 version) from https://www.python.org/ (Whisper claims to run with >3.7 but as of 2023-01-18 some dependencies require >3.7 but <3.11).
    - FFMPEG
        - To install via Scoop (https://scoop.sh/), in powershell run
            - Set-ExecutionPolicy RemoteSigned -Scope CurrentUser
            - irm get.scoop.sh | iex
            - scoop install ffmpeg
        - OPTIONAL/OFFLINE: Follow instructions here: [How to install and use FFMPEG](https://www.thewindowsclub.com/how-to-install-ffmpeg-on-windows-10) and make sure not to skip the part about adding FFMPEG to the Windows PATH variable. 
    -  Git for windows from https://gitforwindows.org/
- Reboot after installing these items.

### Step 2B: Whipser Install (Online Install for Online Use)
-  Open a command prompt and type this command:
    - <code>pip install git+https://github.com/openai/whisper.git</code>
- You may now use Whisper online and no further steps are required. 

### Step 2B: Whipser Install (Online Install for later Offline Use)

- Open a command prompt and type these commands:
  - <code>pip install git+https://github.com/openai/whisper.git</code>
  - <code>pip install blobfile</code>
- Continue to <b>Step 3: Download Other Required Files</b>

### Step 2C: Whipser Install (Offline Install for later Offline Use)
- <b>Option 1:</b> Get the most up to date version of Whisper: 
    - Install Python and Git from Step 1 on an second computer you can connect to the internet and reboot to ensure both are working. 
    - On the ONLINE machine open a command prompt in any empty folder and type the following commands:
        - <code>pip download git+https://github.com/openai/whisper.git</code>
        - <code>pip download blobfile</code>
- <b>Option 2:</b> Download all the necessary files from here [OPENAI-Whisper-20230314 Offline Install Package](https://github.com/nicholasgcotton/WhisperDO/releases/tag/20230314)
- Copy the files to your OFFLINE machine and open a command prompt in that folder where you put the files, and run
    - <code>pip install openai-whisper-20230314.zip</code> (note the date may have changed if you used Option 1 above).
    - <code>pip install blobfile-2.0.2-py3-none-any.whl</code>. (note the version may have changed if you used Option 1 above).
- Continue to <b>Step 3: Download Other Required Files</b>

### Step 3: Download Other Required Files (for Offline Use)
- Download Whisper's Language Model files place them in <code>C:\Users\[Username]\\.cache\whisper</code> Note: If the links are dead updated links can be found at lines 17-27 here: [__init__.py]( https://github.com/openai/whisper/blob/main/whisper/__init__.py) 
    -  [Tiny.En](https://openaipublic.azureedge.net/main/whisper/models/d3dd57d32accea0b295c96e26691aa14d8822fac7d9d27d5dc00b4ca2826dd03/tiny.en.pt)
    -  [Tiny](https://openaipublic.azureedge.net/main/whisper/models/65147644a518d12f04e32d6f3b26facc3f8dd46e5390956a9424a650c0ce22b9/tiny.pt)
    -  [Base.En](https://openaipublic.azureedge.net/main/whisper/models/25a8566e1d0c1e2231d1c762132cd20e0f96a85d16145c3a00adf5d1ac670ead/base.en.pt)
    -  [Base](https://openaipublic.azureedge.net/main/whisper/models/ed3a0b6b1c0edf879ad9b11b1af5a0e6ab5db9205f891f668f8b0e6c6326e34e/base.pt)
    -  [Small.En](https://openaipublic.azureedge.net/main/whisper/models/f953ad0fd29cacd07d5a9eda5624af0f6bcf2258be67c92b79389873d91e0872/small.en.pt)
    -  [Small](https://openaipublic.azureedge.net/main/whisper/models/9ecf779972d90ba49c06d968637d720dd632c55bbf19d441fb42bf17a411e794/small.pt)
    -  [Medium.En](https://openaipublic.azureedge.net/main/whisper/models/d7440d1dc186f76616474e0ff0b3b6b879abc9d1a4926b7adfa41db2d497ab4f/medium.en.pt)
    -  [Medium](https://openaipublic.azureedge.net/main/whisper/models/345ae4da62f9b3d59415adc60127b97c714f32e89e936602e85993674d08dcb1/medium.pt)
    -  [Large-v1](https://openaipublic.azureedge.net/main/whisper/models/e4b87e7e0bf463eb8e6956e646f1e277e901512310def2c24bf0e11bd3c28e9a/large.pt)
    -  [Large-v2](https://openaipublic.azureedge.net/main/whisper/models/81f7c96c852ee8fc832187b0132e569d6c3065a3252ed18e56effd0b6a73e524/large-v2.pt) ([Annoucing the large-v2 model](https://github.com/openai/whisper/discussions/661))
- Download Whisper's vocabulary and encoder files. (Per [issue 1399](https://github.com/openai/whisper/discussions/1399)).
    - Download [Vocab.bpe](https://openaipublic.blob.core.windows.net/gpt-2/encodings/main/vocab.bpe)
    - Download [Encoder.json](https://openaipublic.blob.core.windows.net/gpt-2/encodings/main/encoder.json)
    - Install the files to a folder of your choosing, e.g. <code>C:\Users\[Username]\\.cache\whisper</code>.
    - Update file links in your local copy of openai_public.py which will be installed in your python folder e.g. <code>C:\Users\[UserName]\\AppData\Local\Programs\Python\Python310-32\Lib\site-packagespython3.9/site-packages/tiktoken_ext/openai_public.py</code> to point to where you downloaded the files.
        - Remove the URL <code>"https://openaipublic.blob.core.windows.net/gpt-2/encodings/main/"</code> and replace it with your local copy, e.g. <code>"C:/Users/[Username]/.cache/whisper/vocab.bpe"</code> and <code>"C:/Users/[Username]/.cache/whisper/encoder.json"</code>
       
```python
def gpt2():
    mergeable_ranks = data_gym_to_mergeable_bpe_ranks(
        vocab_bpe_file="C:/Users/nic/.cache/whisper/vocab.bpe",
        encoder_json_file="C:/Users/nic/.cache/whisper/encoder.json",
    )
```
### Alternative Offline Method

See the pre-compiled .exe version of Whisper provided here: [Purfview / Whisper Standalone](https://github.com/Purfview/whisper-standalone-win)

**Comment by Purfview** [comment]
Or just download executable from there: https://github.com/Purfview/whisper-standalone-win

**Reply by nicholasgcotton** [reply]
True and I do recommend that project, but I see this question asked a lot so I figured it would be useful for some to have the proof of concept that it can be done. (Also not that it matters, but when I started figuring this out that other project wasn't an option yet.)

**Reply by 64jcl** [reply]
Looks like a command line executable. Won't that reload the whole model on every run and be highly inefficient?

**Reply by Purfview** [reply]
@64jcl

Do you mean if you want to run on the multiple files?
In Discussions there are bat [aka batch] examples how to run all files in one go: https://github.com/Purfview/whisper-standalone-win/discussions/29

Executable for Faster-Whisper has implementations to do that out of the box.

**Comment by phineas-pta** [comment]
it's a very detailed instruction, but i would advise against the files transfer between pc for offline installation, it's quite tricky

**Reply by nicholasgcotton** [reply]
Oh I agree it's very annoying, but it's unavoidable if you want to set the system up on an offline machine. I had to do this due to security rules at my org, and I figured I might as well share the steps once I'd worked it out.

**Reply by fromoven** [reply]
Hi, I've run into problems installing for offline use because of organisation policy.  I've followed the steps and it craps at the openai-whisper install with the same error: installing build dependencies... pip subprocess... NewConnectionError...  pip._vendor.urlib3...  /simple/setuptools... couldn't find setuptools>=40.8.0.   Setuptools is installed with latest version; I tried updating PIP - use PIP3 instead - specify "setuptools==68.0.0" in the install command.  Same error.  I even tried another computer, re-downloaded the packages with the same result (both times I had Python3.10.  I've look at the forum and see that there's someone else who had the same issue (https://github.com/openai/whisper/discussions/1556) and a reply points back to the steps here.  Any ideas what might be the issue ?  Thanks

**Reply by LqdtrX** [reply]
[fromoven](https://github.com/fromoven), got the same error. Solved: 
1. Download requirements.txt from https://github.com/openai/whisper.git and put it along with other files
2. Run the command: `pip install --no-index --find-links ./ -r requirements.txt`
3. Unpack openai-whisper-20231106.zip and run the command from the corresponding folder: `python setup.py install`

**Comment by IsaacKim9876** [comment]
So by running Whisper api on our own pcs we dont have to pay openAI to use it?

**Reply by Purfview** [reply]
Yeap, it's free.

**Reply by palladinoandrea** [reply]
but also on 2024?

**Reply by docpainting** [reply]
Why

**Reply by mrdarip** [reply]
> Why

you are running whisper on your machine, not in theirs!

**Comment by EnnuiThoughts** [comment]
pip install blobfile-2.0.2-py3-none-any.whl

this returns a no such file exist error

what should i do?

**Reply by Solunexus** [reply]
Check if the files version changed currently blobfile-2.0.2-py3-none-any.whl -> blobfile-2.1.1-py3-none-any.whl

**Comment by imious** [comment]
hi i got this problem with whisper-faster standalone:
![Capture](https://github.com/openai/whisper/assets/148017662/9db71a37-5529-47ad-b46c-17b411fcef39)

**Reply by Purfview** [reply]
Read there: https://github.com/Purfview/whisper-standalone-win/issues/104

**Comment by qingtian305** [comment]
I like this project very much. It works very well. Does whisper-faster have a method for calling programs? I want the engine to be on standby for inference all the time after loading, instead of having to reason from the command line every time, which is slow.

**Reply by qingtian305** [reply]
Also, I wonder if this project can be accelerated with openvino?

**Reply by Purfview** [reply]
> I want the engine to be on standby...

I don't think that's possible with command line executables, it quits when it finishes work on all files.

**Reply by ThreepE0** [reply]
Of course this is possible. Run as a service. Build a simple frontend if you don't like the command line. I'll leave it to you to research how to accomplish this; it's pretty straightforward, and a pretty basic but essential thing to know how to do.

**Comment by gleithoernchen** [comment]
I also installed it on a dedicated server and was wondering if there is another way to use it than with the cmd. Is there an api I can call? This would be easier if I want to use it from other servers in this env.

**Reply by Baka-14** [reply]
Did you find any api to use it ?

**Reply by 64jcl** [reply]
I personally resorted to NodeJS implementation called @xenova/transformers. It is quite easy to use and you just load the model once and can then call on it multiple times. It is easy to rewrite as a server api then:

https://huggingface.co/docs/transformers.js/guides/node-audio-processing

**Comment by StepHaze** [comment]
Is there an instruction on how to install and use Whisper offline (no internet required) for Ubuntu?

**Reply by Purfview** [reply]
I made Whisper executable only for Windows but there is Faster-Whisper executable for Linux: 
https://github.com/Purfview/whisper-standalone-win/releases

It mirrors functionality of Whisper when it's much faster.

**Comment by Sethecientos** [comment]
I have a problem with large and large-v2 models. 

I can use offline everything but them. When I try to select them, whisper tries to connect to the internet to check something, but only with those two models. I redownloaded few times and still happens. 

Any idea?

**Reply by Purfview** [reply]
Do you use the latest version of Whisper?

**Reply by Sethecientos** [reply]
Yes, last version, now I could load large-v2 in python, but in the command line using whisper audio.mp --model large-v2, sometimes fails, sometimes doesn't, large failed everytime. 

GPU is not a issue.

**Reply by Purfview** [reply]
Strange, what if you use "exe" from there? -> https://github.com/Purfview/whisper-standalone-win/releases

**Reply by misutoneko** [reply]
@Sethecientos it might be a good idea to check the checksums of your local model files manually, just in case they've been corrupted somehow (it could even be a hardware problem, like hd dying). The URLs listed [here](https://github.com/openai/whisper/blob/main/whisper/__init__.py) contain the sha256sums for each file.

**Comment by tyomart** [comment]
Hi! You can hate me, but why WhiPser guys? Is there something we should not know?

**Comment by MiguelPerezI** [comment]
Can you add the link for the new large-v3.pt ?

**Comment by ElineDelmarcelle** [comment]
Thank you for the tutorial ! 
Note for beginners like me: the folder /site-packages/tiktoken_ext/openai_public.py will be downloaded in your venv file if you're using a venv. Don't waste time looking for it in the Python folder like I did.

**Comment by Nick222** [comment]
But what about Whisper and Python 3.13.7 on Xubuntu 25.10 ?

**Comment by qingtian305** [comment]
have no idea



ĽĽ
***@***.***
